pkg/testsuite/depositcalculator/depositcalculator.go

```diff
@@ -0,0 +1,335 @@
+func WithMetadataLength(length int) options.Option[Options] {
+	return func(opts *Options) {
+		opts.MetadataLength = length
+	}
+}
+
+// WithStateMetadataLength adds a state metadata feature and adds an entry with an empty key and a dummy value
+// of the given length minus the length of the map length prefix, key length prefix and value length prefix.
+func WithStateMetadataLength(length int) options.Option[Options] {
+	return func(opts *Options) {
+		opts.StateMetadataLength = length
+	}
+}
+
+// WithTagLength adds a tag feature and sets a dummy tag with the given length.
+func WithTagLength(length int) options.Option[Options] {
+	return func(opts *Options) {
+		opts.TagLength = length
+	}
+}
+
+// WithHasNativeToken adds a native token feature.
+func WithHasNativeToken() options.Option[Options] {
+	return func(opts *Options) {
+		opts.HasNativeToken = true
+	}
+}
+
+// WithBlockIssuerKeys adds a block issuer feature and adds the given amount of dummy keys.
+func WithBlockIssuerKeys(keys int) options.Option[Options] {
+	return func(opts *Options) {
+		opts.BlockIssuerKeys = keys
+	}
+}
+
+// WithStakedAmount adds a staking feature and sets the staked amount.
+func WithStakedAmount(amount iotago.BaseToken) options.Option[Options] {
+	return func(opts *Options) {
+		opts.StakedAmount = amount
+	}
+}
+
+// WithImmutableIssuerAddress adds an immutable issuer feature and sets the address.
+func WithImmutableIssuerAddress(address iotago.Address) options.Option[Options] {
+	return func(opts *Options) {
+		opts.ImmutableIssuerAddress = address
+	}
+}
+
+// WithImmutableMetadataLength adds an immutable metadata feature and adds an entry with an empty key and a dummy value
+// of the given length minus the length of the map length prefix, key length prefix and value length prefix.
+func WithImmutableMetadataLength(length int) options.Option[Options] {
+	return func(opts *Options) {
+		opts.ImmutableMetadataLength = length
+	}
+}
+
+func getUnlockConditions[T iotago.UnlockCondition](outputType iotago.OutputType, opts *Options) iotago.UnlockConditions[T] {
+	unlockConditions := make(iotago.UnlockConditions[T], 0)
+
+	// Mandatory address unlocks
+	switch outputType {
+	case iotago.OutputBasic, iotago.OutputAccount, iotago.OutputNFT, iotago.OutputDelegation:
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.AddressUnlockCondition{Address: opts.Address}).(T))
+
+	case iotago.OutputAnchor:
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.StateControllerAddressUnlockCondition{Address: opts.StateControllerAddress}).(T))
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.GovernorAddressUnlockCondition{Address: opts.GovernorAddress}).(T))
+
+	case iotago.OutputFoundry:
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.ImmutableAccountUnlockCondition{Address: &iotago.AccountAddress{}}).(T))
+	}
```

Comment:
There are a bunch of scenarios where this would not yield the correct deposit if the mandatory address(es) is/are not set in the options. It would just silently return a too small value, because the serialized size would be smaller than on an actual output with the address set.
Either panic here, which is fine as long as we keep this as test-only code, or return an error if we want to move it to non-testing code later. I would go with the panics for now, that's simplest.

Reply:
No, it panics if you don't pass the correct addresses for the outputs, because the StorageScore code panics.
It guess it's fine to rely on those panics without any change?